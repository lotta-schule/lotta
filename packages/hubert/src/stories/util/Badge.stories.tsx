import * as React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import { Badge } from '../../badge';

export default {
  title: 'util/Badge',
  Component: Badge,
  argTypes: {},
} as Meta;

const Template: StoryFn = (args) => <Badge {...args} />;

export const Default = {
  render: Template,

  args: {
    value: 12,
  },
};
