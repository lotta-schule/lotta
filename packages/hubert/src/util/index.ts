export * from './Collapse';
export * from './Deletable';
export * from './NoSsr';
export * from './SwipeableViews';
export * from './cssHelpers';
export * from './themeHelpers';
export * from './tooltip';
export * from './useScrollEvent';
export * from './useWindowSize';
