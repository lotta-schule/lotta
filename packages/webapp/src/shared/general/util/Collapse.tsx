import * as React from 'react';
import { motion } from 'framer-motion';

export interface CollapseProps {
  visible: boolean;
  children: React.ReactNode;
}

export const Collapse = ({ visible, children }: CollapseProps) => {
  const variants = {
    visible: {
      height: 'auto',
      opacity: 1,
    },
    hidden: {
      height: 0,
      opacity: 0,
    },
  };
  const state = visible ? 'visible' : 'hidden';
  return (
    <motion.div
      aria-hidden={!visible}
      style={{ overflow: 'hidden' }}
      initial={state}
      animate={state}
      variants={variants}
    >
      {children}
    </motion.div>
  );
};
