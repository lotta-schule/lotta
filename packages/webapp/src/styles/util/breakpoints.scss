$min-xs: 0px;
$min-sm: 600px;
$min-md: 960px;
$min-lg: 1280px;
$min-xl: 1920px;

$max-xs: $min-sm - 1;
$max-sm: $min-md - 1;
$max-md: $min-lg - 1;
$max-lg: $min-xl - 1;
