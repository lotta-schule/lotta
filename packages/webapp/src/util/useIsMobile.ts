import { useMedia } from 'react-use';

export const useIsMobile = () => {
    return useMedia('screen and (max-width: 960px)', false);
};
