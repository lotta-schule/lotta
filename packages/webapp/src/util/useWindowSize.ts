import { useState, useEffect, useCallback } from 'react';

export interface WindowSize {
    innerHeight: number;
    innerWidth: number;
    outerHeight: number;
    outerWidth: number;
}

const getSize = (): WindowSize => ({
    innerHeight: window.innerHeight,
    innerWidth: window.innerWidth,
    outerHeight: window.outerHeight,
    outerWidth: window.outerWidth,
});

export const useWindowSize = () => {
    const [windowSize, setWindowSize] = useState(getSize());

    const handleResize = useCallback(() => {
        setWindowSize(getSize());
    }, []);

    useEffect(() => {
        window.addEventListener('resize', handleResize);
        return () => {
            window.removeEventListener('resize', handleResize);
        };
    }, [handleResize]);

    return windowSize;
};
