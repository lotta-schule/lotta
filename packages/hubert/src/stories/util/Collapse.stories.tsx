import * as React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import { Collapse, CollapseProps } from '../../util';

export default {
  title: 'util/Collapse',
  component: Collapse,
  argTypes: {},
} as Meta;

const Template: StoryFn<CollapseProps> = (args) => (
  <Collapse {...args}>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy
        eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam
        voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet
        clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
        amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
        nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
        sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
        rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
        ipsum dolor sit amet.
      </p>
      <ul>
        <li>Vera; or, The Nihilists (1880) (text)</li>
        <li>The Duchess of Padua (1883) (text)</li>
        <li>Lady Windermere's Fan (1892) (text)</li>
        <li>A Woman of No Importance (1893) (text)</li>
        <li>An Ideal Husband (1895) (text)</li>
        <li>The Importance of Being Earnest (1895) (text)</li>
        <li>Salom√© (1896) Translated from French by Lord Alfred Douglas</li>
        <li>La Sainte Courtisane (Incomplete) (text)</li>
        <li>A Florentine Tragedy (Incomplete) (text)</li>
      </ul>
    </div>
  </Collapse>
);

export const Default = {
  render: Template,

  args: {
    visible: true,
  },
};
