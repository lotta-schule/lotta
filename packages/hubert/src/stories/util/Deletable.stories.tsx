import * as React from 'react';
import { userEvent, within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';
import { Meta, StoryObj } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Avatar } from '../../avatar';
import { Deletable } from '../../util';

export default {
  title: 'util/Deletable',
  component: Deletable,
  argTypes: {},
} as Meta<typeof Deletable>;

export const Default: StoryObj<typeof Deletable> = {
  render: (args) => (
    <Deletable {...args}>
      <Avatar
        src="https://avatars.dicebear.com/api/avataaars/rosa-luxemburg.svg"
        title="Rosa Luxemburg"
      />
    </Deletable>
  ),

  args: {
    onDelete: action('onDelete'),
  },

  play: async ({ canvasElement, initialArgs }) => {
    const fireEvent = userEvent.setup({ delay: 200 });
    const screen = within(canvasElement);

    expect(screen.getByRole('button')).toHaveStyle({ opacity: 0 });

    await fireEvent.click(screen.getByRole('button'));

    expect(initialArgs.onDelete).toHaveBeenCalled();
  },
};
