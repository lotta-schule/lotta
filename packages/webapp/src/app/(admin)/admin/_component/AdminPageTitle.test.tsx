import { render } from 'test/util';
import { AdminPageTitle } from './AdminPageTitle';

vi.mock('@fortawesome/react-fontawesome', async () => ({
  ...vi.importActual('@fortawesome/react-fontawesome'),
  FontAwesomeIcon: vi.fn(() => null),
}));

describe('AdminPageTitle', () => {
  it('should render correctly', () => {
    const screen = render(
      <AdminPageTitle backUrl={'/admin'}>My Page</AdminPageTitle>
    );

    expect(screen.container.firstChild).toMatchSnapshot();
  });
});
