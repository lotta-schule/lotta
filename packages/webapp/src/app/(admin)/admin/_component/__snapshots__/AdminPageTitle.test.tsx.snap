// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdminPageTitle > should render correctly 1`] = `
<div
  data-overlay-container="true"
>
  <div
    class="_root_6a3576 _root_61ca45 _hasScrollableParent_6a3576"
    role="toolbar"
  >
    <h3>
      <a
        aria-current="false"
        class="_root_8f1225 _root_56aed5 _backButton_61ca45 _onlyIcon_56aed5"
        data-variant="default"
        href="/admin"
        role="button"
        type="button"
      >
        <span
          class="_icon_56aed5"
        >
          <span
            class="_root_8d26a2"
          />
        </span>
      </a>
      My Page
    </h3>
  </div>
</div>
`;
